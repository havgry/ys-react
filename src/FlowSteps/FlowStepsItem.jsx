import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const FlowStepsItem = (props) => {
  const {
    children,
    active,
    className,
  } = props

  const wrapperProps = {
    className: classNames(
      'ys-flow-steps__item',
      active && 'ys-flow-steps__item--active',
      className,
    ),
  }

  return (
    <li {...wrapperProps}>
      <span className="ys-flow-steps__label">
        {children}
      </span>
    </li>
  )
}

FlowStepsItem.propTypes = {
  children: PropTypes.string.isRequired,
  active: PropTypes.bool,
  className: PropTypes.string,
}

FlowStepsItem.defaultProps = {
  active: false,
  className: '',
}

export default FlowStepsItem
