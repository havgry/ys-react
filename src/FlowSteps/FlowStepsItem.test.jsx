import React from 'react'
import { shallow, mount as mountComponent } from 'enzyme'

import FlowStepsItem from './FlowStepsItem'

const customClassName = 'gibbon'
const stepLabel = 'step'
const getFlowStepsItem = (children, props) => (
  <FlowStepsItem
    {...props}
  >
    {children}
  </FlowStepsItem>
)

const shallowRender = (children, props) => shallow(getFlowStepsItem(children, props))
const mount = (children, props) => mountComponent(getFlowStepsItem(children, props))

describe('<FlowStepsItem>', () => {
  describe('Props', () => {
    it('Throws propType warning if no `children` prop is passed', () => {
      const component = () => shallowRender()
      expect(component).toThrow()
    })

    it('Throws propType warning if `children` isn\'t a string', () => {
      const component = () => shallowRender(2)
      expect(component).toThrow()
    })

    it('Has correct defaultProps', () => {
      const component = mount()

      const {
        className,
        active,
      } = component.props()

      expect(className).toBe('')
      expect(active).toBe(false)
    })
  })

  describe('Rendering', () => {
    it('Outputs elements with the correct class names', () => {
      const component = shallowRender()

      expect(component.find('.ys-flow-steps__item').exists()).toBe(true)
    })

    it('Custom `className` is being applied to the container', () => {
      const component = shallowRender(stepLabel, {
        className: customClassName,
      })

      expect(component.find(`.ys-flow-steps__item.${customClassName}`).exists()).toBe(true)
    })

    it('Places `children` inside label container', () => {
      const component = shallowRender(stepLabel)

      expect(component.find('.ys-flow-steps__label').text()).toBe(stepLabel)
    })

    it('Passes other generic props to container element', () => {
      const component = shallowRender(stepLabel, {
        id: 'test',
        tabIndex: 0,
      })

      expect(component.props().id).toBe('test')
      expect(component.props().tabIndex).toBe(0)
    })
  })

  describe('Snapshots', () => {
    it('Inactive state (default)', () => {
      const component = shallowRender(stepLabel)

      expect(component).toMatchSnapshot()
    })

    it('Active state', () => {
      const component = shallowRender(stepLabel, {
        active: true,
      })

      expect(component).toMatchSnapshot()
    })

    it('With custom `className`', () => {
      const component = shallowRender(stepLabel, {
        className: customClassName,
      })

      expect(component).toMatchSnapshot()
    })
  })
})
