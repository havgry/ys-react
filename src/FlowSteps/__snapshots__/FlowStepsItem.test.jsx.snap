// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlowStepsItem> Snapshots Active state 1`] = `
<li
  className="ys-flow-steps__item ys-flow-steps__item--active"
>
  <span
    className="ys-flow-steps__label"
  >
    step
  </span>
</li>
`;

exports[`<FlowStepsItem> Snapshots Inactive state (default) 1`] = `
<li
  className="ys-flow-steps__item"
>
  <span
    className="ys-flow-steps__label"
  >
    step
  </span>
</li>
`;

exports[`<FlowStepsItem> Snapshots With custom \`className\` 1`] = `
<li
  className="ys-flow-steps__item gibbon"
>
  <span
    className="ys-flow-steps__label"
  >
    step
  </span>
</li>
`;
